sys/net/link_layer/iQueue_MAC/include/mac_driver.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2015  Zhejiang University, INRIA.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup MAC
+ * @{
+ * @brief   Definition (API) for MAC/RDC drivers implemented in RIOT OS
+ * @author  Shuguo Zhuo <zhuosg@zju.edu.cn>
+ */
+
+#ifndef MAC_DRIVER_H
+#define MAC_DRIVER_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief Return values for MAC/RDC-related functions
+ */
+typedef enum
+{
+    /** @brief No problem encountered */
+    MAC_OK = 0,
+
+    /** @brief Packet transmission deferred; callback function
+               will indicate actual TX outcome in time */
+    MAC_TX_DEFERRED,
+
+    /** @brief A collision has prevented successful operation;
+               user should retry later */
+    MAC_COLLISION = 10,
+
+    /** @brief No ACK packet was returned for the TXed packet;
+               user should retry later */
+    MAC_TX_NO_ACK,
+
+    /** @brief All slots for outgoing packets (for TX) are already taken;
+               user should retry later */
+    MAC_TX_QUEUE_FULL,
+
+    /** @brief The MAC/RDC driver is off */
+    MAC_DRIVER_OFF,
+
+    /** @brief There was not enough memory to perform the wanted operation */
+    MAC_OUT_OF_MEM,
+
+    /** @brief Another, unexpected error has prevented successful operation */
+    MAC_ERROR = 100,
+
+} mac_return_t;
+
+
+/**
+ * @brief Callback function definition for TX status notification
+ *
+ * @param[in] packet_seq_num  Sequence number of the TXed packet concerned
+ * @param[in] callback_param  Arbitrary parameter provided for this callback
+ * @param[in] status          Status of packet transmission
+ * @param[in] num_tx          Number of attempted transmissions for that packet
+ */
+typedef void (* mac_tx_status_callback_t)(uint8_t packet_seq_num,
+        void *callback_param,
+        mac_return_t status,
+        unsigned int num_tx);
+
+/**
+ * @brief Callback function definition for packet RX notification
+ *
+ * @param[in] buf Buffer containing the received packet
+ *                in a raw format (untyped pointer to data).
+ * @param[in] len Length, in bytes, of the raw data (packet)
+ *                to transmit from @c buf @c.
+ * @param[in] rssi Receive Signal Strength Indicator, in dB.
+ * @param[in] lqi Linq Quality Indication.
+ * @param[in] crc_ok @c true @c if this packet's CRC was successfully checked;
+ *                   @c false @c otherwise.
+ */
+typedef void (* mac_rx_packet_notif_t)(void *buf, unsigned int len,
+                                       uint8_t rssi, uint8_t lqi,
+                                       bool crc_ok);
+
+
+/**
+ * @brief Definition of the structure of MAC & RDC protocol drivers for RIOT OS
+ */
+typedef struct MAC_driver
```

Comment:
If I see it correctly `struct MAC_driver` is never used (only the typedef), so you could omit the struct name.